### PR TITLE
Add develop artifacts

### PR DESCRIPTION
**Pull request by AndreyMaslennikov**
In this PR we have extended `bin/ci/az-publish.sh` to create and upload `{develop|release}.txt` and `{develop|release}.json` files to the Azure blob container.

### CODE REVIEW

File: bin/ci/az-publish.sh [hunk 1]
```diff
@@ -41,9 +43,13 @@ VERSION=$(xcrun strings calabash-dylibs/libCalabashFAT.dylib | grep -E "CALABASH
...
 
-az --version
+# Evaluate calabash.framework SHASUM256
+FRAMEWORK_SHASUM256=$(shasum --algorithm 256 ${WORKING_DIR}/calabash.framework/calabash | cut -d " " -f 1)
 
-WORKING_DIR="${BUILD_SOURCESDIRECTORY}"
+# Evaluate dylibFAT SHASUM256
+DYLIBFAT_SHASUM256=$(shasum --algorithm 256 ${WORKING_DIR}/calabash-dylibs/libCalabashFAT.dylib | cut -d " " -f 1)
+
+az --version
```
Comment (maxim-lobanov) [inline bin/ci/az-publish.sh]: Do we really need it? :)

Comment (AndreyMaslennikov) [reply bin/ci/az-publish.sh]: It's interesting for me as well :)
This command has been here before, and no one knows why. Should I remove it?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -90,7 +90,7 @@ jobs:
       AZURE_STORAGE_CONNECTION_STRING: $(azureStorageConnectionString)
       SOURCE_BRANCH: $(Build.SourceBranch)
     displayName: "Publish to Azure Blob Storage"
-    condition: and(succeeded(), eq(variables['Agent.JobName'], variables['mainJob']), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+    condition: and(succeeded(), eq(variables['Agent.JobName'], variables['mainJob']), or(startsWith(variables['Build.SourceBranch'], 'refs/tags/'), startsWith(variables['Build.SourceBranch'], 'develop')))
```
Comment (jmoody) [inline azure-pipelines.yml]: Let's change `mainJob` to `AzurePublishWhen:`

File: bin/ci/az-publish.sh (hunk 1 above)
Comment (jmoody) [reply bin/ci/az-publish.sh]: Ha!  It is left over from when we were exploring publishing to blob storage using the command line tools.

We can remove it or leave it.

File: azure-pipelines.yml (hunk 2 above)
Comment (AndreyMaslennikov) [reply azure-pipelines.yml]: Done

**Pull request merged**